### PR TITLE
[42] introduced system text json as default serializer.

### LINKED ISSUE #42: Utf8 Json serializer AND/OR System.Text.JSON

### PR DESCRIPTION
**Pull request by TimoHeiten**

### CODE REVIEW

File: src/heitech.zer0mqXt.core/Main/Zer0Mq.cs [hunk 1]
```diff
@@ -28,6 +28,14 @@ public IZer0MqBuilder SetSerializer(ISerializerAdapter adapter)
             return this;
         }
 
+        public IZer0MqBuilder UseNewtonsoftJson()
+        {
+            var adapter = new InternalAdapter();
```
Comment (Flash0ver) [inline src/heitech.zer0mqXt.core/Main/Zer0Mq.cs]: Do we need to instantiate a new Adapter here? Could we perhaps just directly `_serializer.SetToNewtonsoft();` on the already existing instance?

File: src/heitech.zer0mqXt.core/Main/Zer0Mq.cs [hunk 2]
```diff
@@ -28,6 +28,14 @@ public IZer0MqBuilder SetSerializer(ISerializerAdapter adapter)
             return this;
         }
 
+        public IZer0MqBuilder UseNewtonsoftJson()
```
Comment (Flash0ver) [inline src/heitech.zer0mqXt.core/Main/Zer0Mq.cs:31]: All tests are now using the new `System.Text.Json` serialization per default.
Perhaps adding one test which confirms that `Newtonsoft.Json` continues to work as expected would be nice.

Comment (TimoHeiten) [reply src/heitech.zer0mqXt.core/Main/Zer0Mq.cs:31]: totally agree. Will do

File: src/heitech.zer0mqXt.core/Main/Zer0Mq.cs (hunk 1 above)
Comment (TimoHeiten) [reply src/heitech.zer0mqXt.core/Main/Zer0Mq.cs]: the member _serializer is of Type ISerializerAdapter which does not offer this method. And a downcast is just bad style here. 

So it seems to me, that this is the best solution without changing too much of the code or introducing unecessary indirections.

Comment (Flash0ver) [reply src/heitech.zer0mqXt.core/Main/Zer0Mq.cs]: Ah, I see. OK

Comment (TimoHeiten) [reply src/heitech.zer0mqXt.core/Main/Zer0Mq.cs]: nvmnd I think I found another sln which might make both of us happy, and I could at the same time utilize the ??= operator :>

Comment (TimoHeiten) [reply src/heitech.zer0mqXt.core/Main/Zer0Mq.cs]: done

File: src/heitech.zer0mqXt.core/Main/Zer0Mq.cs (hunk 2 above)
Comment (TimoHeiten) [reply src/heitech.zer0mqXt.core/Main/Zer0Mq.cs:31]: done

File: src/heitech.zer0mqXt.core/Main/Zer0Mq.cs (hunk 1 above)
Comment (Flash0ver) [reply src/heitech.zer0mqXt.core/Main/Zer0Mq.cs]: Very happy 😉. I love the change indeed.

**Review by Flash0ver — APPROVED** [review]

**Pull request merged**